crates/telio-wg/src/wg.rs

```diff
@@ -612,11 +619,28 @@ impl State {
                     PeerState::Connected,
                 );
 
-                self.send_event(PeerState::Connected, Some(LinkState::Up), peer.clone())
-                    .await?;
+                self.send_event(
+                    PeerState::Connected,
+                    Some(LinkState::Up),
+                    peer.clone(),
+                    false,
+                )
+                .await?;
             }
         }
 
+        let is_direct = |peer: &Peer| {
```

Comment:
:thinking: I can help, but wonder, generally the endpont of `127.0.0.1` is only half-correct way of identifying direct vs relayed peers. E.g. it is, at least theoretically (although in practice it may be hard to achieve), to have another peer on the same computer reachable connected over localhost. 

For this reason, in most (hopefully all) places we are using the endpoints map acquired from `telio-proxy` and comparing wireguard endpoint with that one. 

Therefore I would like recommend a little refactoring. In all cases where `self.send_event(...)` is called, you have access  to variable called "from" which represents the state of WireGuard before the changes. Maybe lets just include `Peer` structure from this `from` wireguard state in the event being sent to the channel. And in the `wg_controller.rs` then we can compare "old" and "new" peer's endpoints against actual `telio-proxy` endpoint :) and determine in this way, whether it was downgraded/upgraded or stayed the same.

Reply:
oh, I meant `device/mod.rs` not `wg_controller.rs`. Sorry for mixup